lib/elixir/lib/io.ex

```diff
@@ -415,8 +415,9 @@ defmodule IO do
   See `IO.getn/3` for a description of return values.
 
   """
-  @spec getn(chardata | String.Chars.t(), pos_integer) :: chardata | nodata
-  @spec getn(device, chardata | String.Chars.t()) :: chardata | nodata
+  @spec getn(prompt | device, count | prompt) :: chardata | nodata
+        when prompt: chardata | String.Chars.t(),
+             count: pos_integer
```

Comment:
The problem with this typespec is that it implies that `chardata, prompt` will work, but it won't. Given typespecs are not used by Dialyzer, they primary purpose is docs, and I think the docs in this case would then become misleading.

Having to choose between both, I would prefer to keep the docs bit accurate.

Reply:
Why do you say typespecs are not used by dialyzer?

Reply:
Dialyzer can verify the typespecs against its own analysis, but they are not used for the analysis itself.